.github/workflows/tools/update-issues/main.js

```diff
@@ -0,0 +1,223 @@
+  // If the original author wants an issue reopened, reopen it.
+  if (issue.closed && !issue.hasLabel(STATUS_ARCHIVED)) {
+    // Important: only load comments if prior filters pass!
+    // If we loaded them on every issue, we could exceed our query quota!
+    await issue.loadComments();
+
+    for (const comment of issue.comments) {
+      body = comment.body.toLowerCase();
+      if (comment.author == issue.author &&
+          comment.createdAt >= issue.closedAt &&
+          body.includes('@shaka-bot') &&
+          (body.includes('reopen') || body.includes('re-open'))) {
+        core.notice(`Found reopen request for issue #${issue.number}`);
+        await issue.reopen();
+        break;
+      }
+    }
+  }
+}
+
+async function manageWaitingIssues(issue) {
+  // Find all waiting issues.
+  if (!issue.closed && issue.hasLabel(STATUS_WAITING)) {
+    const labelAgeInDays = await issue.getLabelAgeInDays(STATUS_WAITING);
+
+    // If an issue has been replied to, remove the waiting tag.
+    // Important: only load comments if prior filters pass!
+    // If we loaded them on every issue, we could exceed our query quota!
+    await issue.loadComments();
+
+    const latestNonTeamComment = issue.comments.find(c => !c.fromTeam);
+    if (latestNonTeamComment &&
+        latestNonTeamComment.ageInDays < labelAgeInDays) {
+      await issue.removeLabel(STATUS_WAITING);
+      return;
+    }
+
+    // If an issue has been in a waiting state for too long, close it as stale.
+    if (labelAgeInDays >= CLOSE_AFTER_WAITING_DAYS) {
+      await issue.postComment(CLOSE_STALE_TEXT);
+      await issue.close();
+    }
+  }
+}
+
+async function cleanUpIssueTags(issue) {
+  // If an issue with the waiting tag was closed, remove the tag.
+  if (issue.closed && issue.hasLabel(STATUS_WAITING)) {
+    await issue.removeLabel(STATUS_WAITING);
+  }
+}
+
+async function pingQuestions(issue) {
+  // If a question hasn't been responded to recently, ping it.
+  if (!issue.closed &&
+      issue.hasLabel(TYPE_QUESTION) &&
+      !issue.hasLabel(STATUS_WAITING)) {
+    // Important: only load comments if prior filters pass!
+    // If we loaded them on every issue, we could exceed our query quota!
+    await issue.loadComments();
+
+    // Most recent ones are first.
+    const lastComment = issue.comments[0];
+    if (lastComment &&
+        lastComment.fromTeam &&
+        // If the last comment was from the team, but not from the OP (in case
+        // the OP was a member of the team).
+        lastComment.author != issue.author &&
+        lastComment.ageInDays >= PING_INACTIVE_QUESTION_DAYS) {
+      await issue.postComment(`@${issue.author} ${PING_QUESTION_TEXT}`);
+      await issue.addLabel(STATUS_WAITING);
+    }
+  }
+}
+
+async function maintainMilestones(issue, nextMilestone, backlog) {
+  // Set or remove milestones based on type labels.
+  if (!issue.closed) {
+    if (issue.hasAnyLabel(LABELS_FOR_NEXT_MILESTONE)) {
```

Comment:
Perhaps it would make sense to add `priority: P4` issues to the backlog, even if they are `type: docs`? We have a few p4 docs issues right now, and they are all marked as backlog.

Reply:
Done.  I've set P3 and P4 as defaulting to the backlog in spite of the type.